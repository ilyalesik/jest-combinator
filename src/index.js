const serializer = require('./serializer')

expect.addSnapshotSerializer(serializer)
