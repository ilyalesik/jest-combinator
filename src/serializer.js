
const KEY = '__jest-combinator__';

const getNodes = (node, nodes = []) => {
    if (typeof node === 'object') {
        nodes.push(node)
    }

    if (node.children) {
        Array.from(node.children).forEach(child => getNodes(child, nodes))
    }

    return nodes
}

const markNodes = nodes => nodes.forEach(node => (node[KEY] = true))

const serializer = {
    test(val) {
        return (
            val &&
            !val[KEY] &&
            (val.$$typeof === Symbol.for('react.test.json') ||
                (global.Element && val instanceof global.Element))
        )
    },

    print(val, print) {
        console.log(val);
        const nodes = getNodes(val);
        markNodes(nodes);
        return print(val);
    },
};

module.exports = serializer;
