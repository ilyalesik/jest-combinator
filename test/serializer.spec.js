import React from "react";
import renderer from 'react-test-renderer'
import "../src/index"

describe("serializer", () => {
    it("simple", () => {
        const component = (
            <div>component</div>
        );

        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    })
})
