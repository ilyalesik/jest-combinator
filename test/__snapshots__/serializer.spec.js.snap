// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializer simple 1`] = `
<div>
  component
</div>
`;
